src/panels/lovelace/editor/config-elements/hui-entities-card-editor.ts

```diff
@@ -105,28 +87,15 @@ export class HuiEntitiesCardEditor extends LitElement
       return html``;
     }
 
-    if (this._editRowConfig) {
+    if (this._subElementEditorConfig) {
       return html`
-        <hui-detail-editor-base
+        <hui-sub-element-editor
           .hass=${this.hass}
-          .guiModeAvailable=${this._editRowGuiModeAvailable}
-          .guiMode=${this._editRowGuiMode}
-          @toggle-gui-mode=${this._toggleMode}
+          .config=${this._subElementEditorConfig}
```

Comment:
This doesn't work, as this object will not update when it is changed in the editor. So changing from the visual editor to the code editor will always use to the old config. (and the `hui-sub-element-editor` should not manage the config, it should be passed on update)

Reply:
Not sure I understand. The config is updated in the `_handleSubElementChanged`
```ts
this._subElementEditorConfig!.elementConfig = value;
```

Reply:
`@config-changed` event that happens from the `hui-row-element-editor` and the `hui-headerfooter-element-editor` actually passes through the `hui-sub-element-editor` and into the card editor.

Reply:
`hui-sub-element-editor.ts`
```ts
private _handleConfigChanged(ev: CustomEvent): void {
  this._guiModeAvailable = ev.detail.guiModeAvailable;
}
```

Reply:
Try editing something in the UI editor and then change it to the code editor, the code editor config will not be updated.